Source/OxyPlot.SkiaSharp.Avalonia/DoubleBuffered/PlotRenderer.cs

```diff
@@ -0,0 +1,225 @@
+        private Exception renderException;
+        private readonly object frontBufferLock = new();
+        private SKBitmap frontBuffer;
+        private SKBitmap backBuffer;
+        private CancellationTokenSource renderCancellationTokenSource;
+        private readonly AutoResetEvent renderRequiredEvent = new(false);
+        private readonly Mutex renderLoopMutex = new();
+
+        public PlotView PlotView { get; } = parent;
+
+        /// <summary>
+        /// Notifies the <see cref="PlotRenderer"/> that a re-render is required.
+        /// </summary>
+        public void RequestRender()
+        {
+            this.renderRequiredEvent.Set();
+        }
+
+        /// <inheritdoc />
+        public override void Render(DrawingContext context)
+        {
+            if (this.renderException is not null)
+            {
+                var exceptionText = new FormattedText(
+                    this.renderException.ToString(),
+                    CultureInfo.CurrentCulture,
+                    CultureInfo.CurrentCulture.TextInfo.IsRightToLeft ? FlowDirection.RightToLeft : FlowDirection.LeftToRight,
+                    Typeface.Default,
+                    10,
+                    Brushes.Black);
+
+                context.DrawText(exceptionText, new Point(20, 20));
+                return;
+            }
+
+            using var drawOperation = new DoubleBufferDrawOperation(new Rect(0, 0, this.Bounds.Width, this.Bounds.Height), this);
+            context.Custom(drawOperation);
+        }
+
+        private void BufferSwitch()
+        {
+            lock (this.frontBufferLock)
+            {
+                (this.frontBuffer, this.backBuffer) = (this.backBuffer, this.frontBuffer);
+            }
+        }
+
+        /// <summary>
+        /// Makes sure that the backbuffer is initialized with the required size.
+        /// </summary>
+        /// <param name="size">The size (in device-independent pixels).</param>
+        /// <returns>The DPI scaling factor.</returns>
+        private double EnsureBackBuffer(Size size)
+        {
+            var dpiScale = TopLevel.GetTopLevel(this)?.RenderScaling ?? 1;
+            var width = (int)(size.Width * dpiScale);
+            var height = (int)(size.Height * dpiScale);
+            if (this.backBuffer is null || this.backBuffer.Width != width || this.backBuffer.Height != height)
+            {
+                this.backBuffer?.Dispose();
+                this.backBuffer = new SKBitmap(width, height, SKColorType.Rgba8888, SKAlphaType.Premul);
+            }
+
+            return dpiScale;
+        }
+
+        /// <summary>
+        /// This loop runs until canceled and updates and renders the plot if required.
+        /// </summary>
+        private void RenderLoop(CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                this.renderRequiredEvent.WaitOne();
```

Comment:
I don't know if the guidance has changed on this (and I'm just glancing at the PR, not taken a proper look yet, so may be talking nonsense), but I don't think locking up a thread like this is ideal. Using `https://learn.microsoft.com/en-us/dotnet/api/system.threading.semaphoreslim.waitasync?view=net-8.0` or similar would allow it to run on the thread-pool and would avoid thread starving

Reply:
You are correct of course, and usually I would prefer not to block a thread here.
However `SemaphoreSlim` does not really cover the use-case we need from the `AutoResetEvent` here: It's possible that updates are requested more often than they can be performed, and I don't think this scenario really works with a semaphore. Usually I'd like to use `AsyncAutoResetEvent` from `AsyncEx` for this purpose, but I didn't want to bring in this dependency just for this. Shame that .NET doesn't provide stuff like this out of the box...

Reply:
`SemaphoreSlim` has always been my go-to for this; as you say, it's annoying that there's no equivalent method on `AutoResetEvent`. Not familiar enough with the code to say for sure, but if we move the CAS logic earlier it should implicitly solve that problem: I'm happy to look into this in more detail.

Reply:
I am not quite sure how this could be made to work with a `SemaphoreSlim`, but I would be curious to find out if that is possible. So you are quite welcome to look into this if you are motivated :)

Reply:
Idea is that you use a dirty flag to decide whether the code should actually run (it already does this for update vs render), then the semaphore will just spin quickly until the count is 0. I'll stop posting inane commentary here, and hopefully take a proper look at the whole PR over the weekend ;)

Reply:
There is definitely room for improvement here, I just realized that the `isRenderRequired` flag isn't actually used anywhere. And thinking about it, I guess this makes sense, because in the double-buffer configuration we only need to render when something actually has changed in the PlotModel...